logbook-api/src/main/java/org/zalando/logbook/Strategy.java

```diff
@@ -119,4 +121,8 @@ default void write(final Correlation correlation, final HttpRequest request, fin
         sink.write(correlation, request, response);
     }
 
+    default RequestAttributesExtractor getRequestAttributesExtractor() {
```

Comment:
Why attributes extractors should be coupled with strategies and not be independent concept? If I want to use `StatusAtLeastStrategy` I should be able to use JWT attributes at the same time. 

Suggestion: add AttributesExtractor as a separate property in the `LogbookCreator`.

Reply:
They are not coupled; please see [this example](https://github.com/msdousti/LogbookJwtTester/blob/45d2b385fb937e44f13ed660fcc8648fd2faa48b/src/main/java/org/example/Main.java#L66-L77).

It is also a possibility to add it to `LogbookCreator` and I considered it; but having a single Strategy made more sense to me.

Of course I'm open for changing it - Let's discuss tomorrow.

Reply:
> They are not coupled in the sense that you can easily mix

It may be easy (actually debatable, but let's say it is), but I wouldn't call it simple in the sense of Rich Hickey's [Simple made Easy](https://www.youtube.com/watch?v=LKtk3HCgTa8) talk.

> but having a single Strategy made more sense to me.

I'm with @kasmarian on this. `Strategy` and `AttributesExtractor` are are two unrelated concepts, they shouldn't depend or coupled together like this.

Reply:
Hi Willi,

Thanks a lot for the review! It's really encouraging, and I'll apply them as soon as possible 👍 
I talked with Karen yesterday about how to proceed with the rest of this PR, and to add the HttpAttributes to the response as well. We have several choices:

- Have separate strategy interfaces for extracting them from request and from response, or a single interface having two methods.
- The method to extract HttpAttributes from the response can only take response as parameter, or both request and response.
- We can allow an array/list of such strategies to be registered during the creation of Logbook, or only a single strategy that combines them all.

Suggestions are very welcome!

Reply:
> * Have separate strategy interfaces for extracting them from request and from response, or a single interface having two methods.

I'd go with a single interface and two methods both with a default implementation that returns an empty list of attributes maybe?

> * The method to extract HttpAttributes from the response can only take response as parameter, or both request and response.

I'd add both, giving implementations more information, e.g. only extract response attributes from those responses where the request matches a certain criterion.

> * We can allow an array/list of such strategies to be registered during the creation of Logbook, or only a single strategy that combines them all.

I'd go with a `CompositeAttributesExtractor` implementation that holds a list of extractors and is dealing with the loop internally. It should be an implementation detail though and during Logbook creation the user should be able to just call the corresponding builder method multiple times. I think we do the same approach with filters.

Reply:
Thanks a lot Willi, I'll follow your excellent suggestions. 👏

Reply:
Applied all suggestions.